src/js/components/SelectMultiple/SelectMultipleContainer.js

```diff
@@ -0,0 +1,614 @@
+      (event) => {
+        if (!onSearch) {
+          const nextActiveIndex = options.findIndex((e) => {
+            let label;
+            if (typeof e === 'object') {
+              label = e.label || applyKey(e, labelKey);
+            } else {
+              label = e;
+            }
+            return (
+              typeof label === 'string' &&
+              label.charAt(0).toLowerCase() === event.key.toLowerCase()
+            );
+          });
+
+          if (nextActiveIndex >= 0) {
+            event.preventDefault();
+            setActiveIndex(nextActiveIndex);
+            setKeyboardNavigation(true);
+          }
+        }
+        if (onKeyDown) {
+          onKeyDown(event);
+        }
+      },
+      [onKeyDown, options, onSearch, labelKey],
+    );
+
+    const onActiveOption = useCallback(
+      (index) => () => {
+        if (!keyboardNavigation) setActiveIndex(index);
+      },
+      [keyboardNavigation],
+    );
+
+    const onSelectOption = useCallback(
+      (event) => {
+        if (
+          !checkDisabled(
+            activeIndex,
+            disabled,
+            disabledKey,
+            options,
+            valueKey || labelKey,
+          ) &&
+          activeIndex >= 0 &&
+          activeIndex < options.length
+        ) {
+          event.preventDefault(); // prevent submitting forms
+          selectOption(activeIndex)(event);
+        }
+      },
+      [
+        activeIndex,
+        selectOption,
+        options,
+        disabled,
+        disabledKey,
+        valueKey,
+        labelKey,
+      ],
+    );
+
+    const customSearchInput = theme.select.searchInput;
+    const SelectTextInput = customSearchInput || TextInput;
+    const selectOptionsStyle = theme.select.options
+      ? {
+          ...theme.select.options.box,
+          ...theme.select.options.container,
+        }
+      : {};
+
+    // handle when limit is reached
+    useEffect(() => {
+      if (value.length === limit) {
+        const newDisabled = [...disabledProp];
+        // disable everything that is not selected
+        const originallyDisabled = (index) => {
+          const option = allOptions[index];
+          let result;
+          if (disabledKey) {
+            result = applyKey(option, disabledKey);
+          } else if (Array.isArray(disabledProp)) {
+            if (typeof disabledProp[0] === 'number') {
+              result = disabledProp.indexOf(index) !== -1;
+            } else {
+              result =
+                disabledProp.indexOf(
+                  getOptionValue(index, options, valueKey || labelKey),
+                ) !== -1;
+            }
+          }
+          return result;
+        };
+        for (let i = 0; i < options.length; i += 1) {
+          if (!isSelected(i) && !originallyDisabled(i)) {
+            newDisabled.push(options[i]);
+          }
+        }
+        if (usingKeyboard)
+          setShowA11yLimit('Selected. Maximum selection limit reached.');
```

Comment:
This string message seems to read weirdly

Reply:
Should it just be "Maximum selection limit reached."

Reply:
Yeah so this message gets read when an option is selected and the number of selected options is now equal to the limit. I wanted to give some confirmation that the option was actually selected. I was thinking if the message just read 'Maximum selection limit reached' the user might be confused as to whether their option was selected or not.